### PR TITLE
Fix Spark bool comparison function

### PR DESCRIPTION
**Pull request by zhli1142015**
Fix bug in https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/Comparisons.cpp#L106
```
      rows.applyToSelected([&](vector_size_t i) {
        flatResult->set(
            i, cmp(bits::isBitSet(rawA, i), bits::isBitSet(rawA, i)));
      });
```

### CODE REVIEW

File: velox/functions/sparksql/Comparisons.cpp [hunk 1]
```diff
@@ -43,10 +43,11 @@ class ComparisonFunction final : public exec::VectorFunction {
 
     if (args[0]->isFlatEncoding() && args[1]->isFlatEncoding()) {
       // Fast path for (flat, flat).
-      auto rawA = args[0]->asUnchecked<FlatVector<T>>()->mutableRawValues();
-      auto rawB = args[1]->asUnchecked<FlatVector<T>>()->mutableRawValues();
-      rows.applyToSelected(
-          [&](vector_size_t i) { flatResult->set(i, cmp(rawA[i], rawB[i])); });
+      auto rawLeft = args[0]->asUnchecked<FlatVector<T>>()->mutableRawValues();
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply velox/functions/sparksql/Comparisons.cpp:46]: Yes, ok to me.

File: velox/functions/sparksql/Comparisons.cpp [hunk 2]
```diff
@@ -103,7 +103,7 @@ class BoolComparisonFunction final : public exec::VectorFunction {
                       ->template mutableRawValues<uint64_t>();
       rows.applyToSelected([&](vector_size_t i) {
         flatResult->set(
-            i, cmp(bits::isBitSet(rawA, i), bits::isBitSet(rawA, i)));
+            i, cmp(bits::isBitSet(rawA, i), bits::isBitSet(rawB, i)));
```
Comment (mbasmanova) [inline velox/functions/sparksql/Comparisons.cpp:106]: Wow!!! Nice finding.

File: velox/functions/sparksql/Comparisons.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhli1142015) [reply velox/functions/sparksql/Comparisons.cpp]: Updated, thanks.

File: velox/functions/sparksql/tests/ComparisonsTest.cpp [hunk 3]
```diff
@@ -65,6 +65,15 @@ class ComparisonsTest : public SparkFunctionBaseTest {
       std::optional<T> b) {
     return evaluateOnce<bool>("greaterthanorequal(c0, c1)", a, b);
   }
+
+  void runAndCompare(
+      const std::string& funcStr,
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply velox/functions/sparksql/tests/ComparisonsTest.cpp]: Updated, thanks.

File: velox/functions/sparksql/tests/ComparisonsTest.cpp [hunk 4]
```diff
@@ -65,6 +65,15 @@ class ComparisonsTest : public SparkFunctionBaseTest {
...
+      std::vector<VectorPtr>& input,
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply velox/functions/sparksql/tests/ComparisonsTest.cpp]: Updated, thanks.

**Review by mbasmanova — APPROVED** [review]
@zhli1142015 Thank you for the fix.

**Pull request closed without merging**